actionpack/lib/action_dispatch/middleware/flash.rb

```diff
@@ -241,10 +234,19 @@ def keep(k = nil)
         k ? self[k] : self
       end
 
+      # Immediately deletes the single flash entry. Use this method when you want remove the message within the current action. See also <tt>discard</tt>.
+      def delete(key)
+        key = key.to_s
+        @discard.delete key
+        @flashes.delete key
+        self
+      end
+
       # Marks the entire flash or a single flash entry to be discarded by the end of the current action:
       #
       #     flash.discard              # discard the entire flash at the end of the current action
       #     flash.discard(:warning)    # discard only the "warning" entry at the end of the current action
+      # Use this method when you want to display the message in the current action but not in the next one. See also <tt>delete</tt>.
```

Comment:
```suggestion

      # Use this method when you want to display the message in the current action but not in the next one. See also <tt>delete</tt>.
```

Newlines between the code sample and other content :pray: